utils/scripts/patch_files.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+# Removes static libraries and and prefixes library names matching
+# lib${prefix}* with libdaos${prefix}*.  Modifies the libraries
+# to change dependencies accordingly.
+bindir="$1"; shift
+libdir="$1"; shift
+
+mkdir -p "${libdir}/daos_internal"
+#create links first
+readarray -d '' links < <(find "${libdir}" -type l -print0)
+for file in "${links[@]}"; do
+  target="$(readlink "${file}")"
+  echo "Checking to see if link ${file} -> ${target} needs fixing"
+  lib="$(basename "${file}")"
+  for prefix in "$@"; do
+      newtarget="${target//lib${prefix}/libdaos${prefix}}"
+      echo "Checking ${newtarget}"
+      if [ "${newtarget}" != "${target}" ]; then
+        # create a link from original filename in daos_internal to the
+        # target.  This way link time name is unchanged for DAOS build.
+        echo "Creating link ${libdir}/daos_internal/${lib} -> ${newtarget}"
+        ln -s ../"${newtarget}" "${libdir}/daos_internal/${lib}"
+        newlib="${lib//lib${prefix}/libdaos${prefix}}"
```

Comment:
double "//" is that necessary for escaping/recursive substitution?

Reply:
Honestly, I would use sed here but shellcheck suggested using these substitutions and I haven't done the research to fully grok them

Reply:
It took me a day to fix all of the problems after I fixed the shellcheck warnings